handlers/userPayments.js

```diff
@@ -9,29 +9,69 @@ function userPayments(request, reply) {
     return reply('You arent allowed to do this');
   }
 
+  if (request.method === 'get') getHandler(request, reply, userId);
+  else if (request.method === 'post') postHandler(request, reply, userId);
+}
+
+
+function getHandler(request, reply, userId){
+    pool.getConnection(function(err, connection) {
+      connection.query({
+        sql: 'SELECT * FROM payments y, properties p ' +
+             'WHERE y.propertyId = p.id AND y.tenantId = ?',
+        values: [userId],
+      }, function(err, results) {
+        connection.release();
+        if (err) return reply(err);
+
+        // Put the result set in a more consumable form
+        var massagedResults = results.map(function(row) {
+          return {
+            id: row.id,
+            date: row.date,
+            property: row.street + ', ' + row.suburb,
+            amount: row.amount
+          };
+        });
+
+        reply(massagedResults);
+      });
+    });
+}
+
+function postHandler(request, reply, userId) {
+  var payload = request.payload;
+
   pool.getConnection(function(err, connection) {
     connection.query({
-      sql: 'SELECT * FROM payments y, properties p ' +
-           'WHERE y.propertyId = p.id AND y.tenantId = ?',
-      values: [userId],
+      sql: 'SELECT id FROM properties WHERE tenantId = ?',
+      values: [userId]
     }, function(err, results) {
-      connection.release();
-      if (err) return reply(err);
-
-      // Put the result set in a more consumable form
-      var massagedResults = results.map(function(row) {
-        return {
-          date: row.date,
-          property: row.street + ', ' + row.suburb,
-          amount: row.amount
-        };
-      });
+      if (err) {
+        connection.release();
+        return reply(err);
+      }
+      var propertyId = results[0].id;
+
+      connection.query({
+        sql: 'INSERT INTO payments (amount, tenantId, propertyId)' +
+             'VALUES (?, ?, ?)',
+        values:[
```

Comment:
Always put a space after a property in an object. Eg.

`values: [...]` rather than `values:[...]`